sdk/storage/inc/shares/protocol/share_rest_client.hpp

```diff
@@ -2158,9 +2088,12 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
           request.AddQueryParameter(
               Details::c_QueryTimeout, std::to_string(createOptions.Timeout.GetValue()));
         }
-        for (const auto& pair : createOptions.Metadata.GetValue())
+        if (createOptions.Metadata.HasValue())
```

Comment:
wait a minute, we already made a conclusion that metadata isn't optional. What're you doing here?

Reply:
Oops, forgot it, sorry.